src/Http/ResponseBuilder.php

```diff
@@ -54,15 +69,19 @@ public function __construct($responseFactory)
      *
      * @param  int|null $statusCode
      * @param  array    $headers
+     * @param  bool     $successFlag
      * @return \Illuminate\Http\JsonResponse
      */
-    public function respond(int $statusCode = null, array $headers = []):JsonResponse
+    public function respond(int $statusCode = null, array $headers = [], bool $successFlag = true):JsonResponse
```

Comment:
I guess a better option would be to override the `respond` method in each of the response builders, set the success flag and then call on the parent method?